app/controllers/posts_controller.rb

```diff
@@ -5,10 +5,14 @@ def index
   end
 
   def show
-    @user = User.find_by_id(params[:user_id]) || not_found
-    @posts = @user.posts
-    @post = @posts.find_by_id(params[:id]) || not_found
-    @comments = @post.comments
+    # use includes to avoid n+1 queries
+    # find the user by id and include posts and comments
+    @user = User.includes(:posts, posts: [:comments])
+      .where(posts: { id: params[:id] })
+      .find_by_id(params[:user_id]) || not_found
+
+    # find the post by id
+    @post = @user.posts.find_by_id(params[:id]) || not_found
   end
 
```

Comment:
![image](https://github.com/tsohleDev/weblog/assets/93039359/2981e0b9-41bf-4356-8b42-d2aa77ffc759)

**[OPTIONAL]**

- [ ] It would be nice if you make all your commit message capitalized to make your pull request professional and descriptive 👍

Reply:
Thank you sir 👍

Reply:
Will do